src/main/java/org/spin/grpc/service/field/payment/PaymentInfoConvert.java

```diff
@@ -0,0 +1,144 @@
+package org.spin.grpc.service.field.payment;
+
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import org.compiere.util.Env;
+import org.adempiere.core.domains.models.I_C_Payment;
+import org.compiere.model.MPayment;
+import org.spin.backend.grpc.field.payment.PaymentInfo;
+import org.spin.service.grpc.util.value.NumberManager;
+import org.spin.service.grpc.util.value.TimeManager;
+import org.spin.service.grpc.util.value.ValueManager;
+
+public class PaymentInfoConvert {
+	public static PaymentInfo.Builder convertPaymentInfo(ResultSet rs) throws SQLException {
+		PaymentInfo.Builder builder = PaymentInfo.newBuilder();
+		if (rs == null) {
+			return builder;
+		}
+
+		int paymentId = rs.getInt(
+			I_C_Payment.COLUMNNAME_C_Payment_ID
+		);
+
+		MPayment payment = new MPayment(Env.getCtx(), paymentId, null);
+
+
+		builder.setId(
+				rs.getInt(
+					I_C_Payment.COLUMNNAME_C_Payment_ID
+				)
+			)
+			.setUuid(
+				ValueManager.validateNull(
+					rs.getString(
+                        I_C_Payment.COLUMNNAME_UUID
+                    )
+				)
+			)
+			.setDisplayValue(
+				ValueManager.validateNull(
+					payment.getDisplayValue()
+				)
+			)
+			.setBankAccount(
+				ValueManager.validateNull(
+					rs.getString(
+						"BankAccount"
+					)
+				)
+			)
+			.setBusinessPartner(
+				ValueManager.validateNull(
+					rs.getString("BusinessPartner")
+				)
+			)
+			.setDatePayment(
+				TimeManager.convertDateToValue(
+					rs.getTimestamp(
+						I_C_Payment.COLUMNNAME_DateTrx
+					)
+				)
+			)
+			.setDocumentNo(
+				ValueManager.validateNull(
+					rs.getString(
+						I_C_Payment.COLUMNNAME_DocumentNo
+					)
+				)
+			)
+			.setIsReceipt(
+				rs.getBoolean(
+					I_C_Payment.COLUMNNAME_IsReceipt
+				)
+			)
+			.setCurrency(
+				ValueManager.validateNull(
+					rs.getString("Currency")
+				)
+			)
+			.setPayAmt(
+				NumberManager.getBigDecimalToString(
+					rs.getBigDecimal(
+						I_C_Payment.COLUMNNAME_PayAmt
+					)
+				)
+			)
+			.setConvertedAmount(
+				NumberManager.getBigDecimalToString(
+					rs.getBigDecimal("currencyBase")
+				)
+			)
+			.setDiscountAmt(
+				NumberManager.getBigDecimalToString(
+					rs.getBigDecimal(
+						I_C_Payment.COLUMNNAME_DiscountAmt
+					)
+				)
+			)
+			.setWriteOffAmt(
+				NumberManager.getBigDecimalToString(
+					rs.getBigDecimal(
+						I_C_Payment.COLUMNNAME_WriteOffAmt
+					)
+				)
+			)
+			
+			.setIsAllocated(
+				rs.getBoolean(
+					I_C_Payment.COLUMNNAME_IsAllocated
+				)
+			)
+			// .setIsDelivered(
+			// 	rs.getBoolean(
+			// 		"isDelivered"
+			// 	)
```

Comment:
remove this coments

Reply:
https://github.com/solop-develop/adempiere-grpc-server/pull/781/commits/a4f5724b9ad3dbae04752282373d8492dd6f7494